### PR TITLE
Remove Microsoft.Win32 dependency from System.Private.Xml

### PR DESCRIPTION
**Pull request by sepidehMS**
Fixes https://github.com/dotnet/corefx/issues/15809

cc: @danmosemsft  @jkotas @weshaggard 

FYI: @joperezr (https://github.com/dotnet/corefx/issues/15966)

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs [hunk 1]
```diff
@@ -753,50 +751,6 @@ internal XmlReader AddConformanceWrapper(XmlReader baseReader)
```
Comment (jkotas) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: Delete `s_enableLegacyXmlSettings`

File: src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs [hunk 2]
```diff
@@ -753,50 +751,6 @@ internal XmlReader AddConformanceWrapper(XmlReader baseReader)
 
         internal static bool EnableLegacyXmlSettings()
         {
```
Comment (jkotas) [inline src/System.Private.Xml/src/System/Xml/Core/XmlReaderSettings.cs]: Is there a good reason for keeping this method around? It is called from just 5 places that are straightforward to fix.

File: src/System.Private.Xml/src/System.Private.Xml.csproj [hunk 3]
```diff
@@ -119,7 +119,6 @@
     <Compile Include="System\Xml\Core\XmlSubtreeReader.cs" />
     <Compile Include="System\Xml\Core\XmlSubtreeReaderAsync.cs" />
     <Compile Include="System\Xml\Core\XmlTextEncoder.cs" />
-    <Compile Include="System\Xml\Core\BinaryCompatibility.cs" />
```
Comment (jkotas) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: Also delete reference to Microsoft.Win32.Registry at the bottom of the file

**Comment by sepidehMS** [comment]
Sorry I submitted an incomplete version of my changes. Will fix it right now.

File: src/System.Private.Xml/src/System.Private.Xml.csproj (hunk 3 above)
Comment (jkotas) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: (You can also convert System.Threading.Tasks.Extensions from project reference to contract reference like all the other ones while you are on it.)

Comment (sepidehMS) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: Done.

Comment (danmosemsft) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: Delete of the .cs file?

Comment (sepidehMS) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: Ah, I'm being so careless. Done.

Comment (sepidehMS) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: Looks like there is no contract for System.Threading.Tasks.Extensions. Keeping the project reference.

Comment (jkotas) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: ok

Comment (weshaggard) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: It doesn't need a package reference for System.Threading.Tasks.Extensions. If you doesn't work with a reference then there is likely something else wrong that we might need to fix. Can you tell me what you error you get when you change it to a Reference?

Comment (sepidehMS) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: @weshaggard Doesn't it need a reference assembly for it? 
`System/Xml/Core/XmlTextReaderImplAsync.cs(3363,17): error CS0246: The type or namespace name 'ValueTask<>' could not be found (are you missing a using directive or an assembly reference?)`

Comment (weshaggard) [inline src/System.Private.Xml/src/System.Private.Xml.csproj]: ah yes it doesn't have an explicit ref. We have another work item tracking how to fix those, for now keep the ProjectReference. Thanks for trying.

**Pull request merged**